src/cmd_exec.rs

```diff
@@ -0,0 +1,195 @@
+extern crate base64;
+
+use super::*;
+use keylime_error;
+use std::env;
+use std::process::Command;
+use std::process::Output;
+use std::str;
+use std::thread;
+use std::time::Duration;
+use std::time::SystemTime;
+
+const MAX_TRY: usize = 10;
+const RETRY_SLEEP: Duration = Duration::from_millis(50);
+const TPM_IO_ERROR: i32 = 5;
+const RETRY: usize = 4;
+static EMPTYMASK: &'static str = "1";
+
+/*
+ * Input:
+ *     command: command to be executed
+ *     output_path: file output location
+ * return:
+ *     execution return output and file output
+ *     KeylimeTpmError
+ *
+ * Set up execution envrionment to execute tpm command through shell commands
+ * and return the execution result in a tuple. Based on the latest update of
+ * python keylime this function implement the functionality of cmd_exec
+ * script in the python keylime repo. RaiseOnError, return code and lock are
+ * dropped due to different error handling in Rust. Returned output string are
+ * preprocessed to before returning for code efficient.
+ */
+pub fn run<'a>(
+    command: String,
+    output_path: Option<&str>,
+) -> Result<(String, String), keylime_error::KeylimeTpmError> {
+    let mut file_output = String::new();
+    let mut output: Output;
+
+    // tokenize input command
+    let words: Vec<&str> = command.split(" ").collect();
```

Comment:
Aha, now I see #84.

Reply:
I can see the problem now, I used the wrong "Resolves" node and put a PR instead of the issue I had open:

https://github.com/keylime/rust-keylime/issues/83

Let me correct 82 to 83 in the commit message and remove the change on line 42 where we swap the forward slash for a whitespace (which is covered in issue #83)